src/coreclr/src/zap/zapinfo.cpp

```diff
@@ -2171,8 +2171,10 @@ DWORD FilterNamedIntrinsicMethodAttribs(ZapInfo* pZapInfo, DWORD attribs, CORINF
         fTreatAsRegularMethodCall = fIsGetIsSupportedMethod && fIsPlatformHWIntrinsic;
 
 #if defined(TARGET_ARM64)
-        // On Arm64 AdvSimd ISA is required by CoreCLR, so we can expand Vector64<T> and Vector128<T> methods.
-        fTreatAsRegularMethodCall |= !fIsPlatformHWIntrinsic && fIsHWIntrinsic && (strcmp(className, "Vector64`1") != 0) && (strcmp(className, "Vector128`1") != 0);
+        // On Arm64 AdvSimd ISA is required by CoreCLR, so we can expand Vector64<T> and Vector128<T> generic methods (e.g. Vector64<byte>.get_Zero)
+        // as well as Vector64 and Vector128 methods (e.g. Vector128.CreateScalarUnsafe).
+        fTreatAsRegularMethodCall |= !fIsPlatformHWIntrinsic && fIsHWIntrinsic
+            && (strstr(className, "Vector64") != className) && (strstr(className, "Vector128") != className);
```

Comment:
This condition could be checked more accurately by 
```c++
(strcmp(className, "Vector64") != 0) && (strcmp(className, "Vector64`1") != 0) && (strcmp(className, "Vector128") != 0) && (strcmp(className, "Vector128`1") != 0)
```

Reply:
why `strstr` rather than `strncmp`?

Reply:
> why `strstr` rather than `strncmp`?

@tannergooding What do you mean?

Reply:
`strstr` checks the entire string and is more complex, so if you have `strstr(className, "Vector64")` and `className` happend to be "SomeTextVector64", it wouldn't exit on `S` and would instead keep searching and match `Vector64` at the end, returning the new index.

`strncmp` allows you to check for `"Vector64"` or `"Vector128"`, from the start of the string and terminate at a given digit count. This means you know it starts with exactly that text and if desired you can easily check if the returned index is `'\0'` or if it is `"``1"`)

Reply:
Makes sense

Reply:
@tannergooding I did as you suggested - replaced `strstr` with `strncmp`. PTAL